### PR TITLE
Issue #6771: update xdoc files to use violation term instead of error

### PR DESCRIPTION
**Pull request by romani**
Issue #6771 , this is not a final fix, just fixes over `xdoc` folder.

please review https://github.com/checkstyle/checkstyle/issues/6771#issuecomment-524585694

most interesting changes are at the bottom of github change page.

### CODE REVIEW

File: src/xdocs/writinglisteners.xml.vm [hunk 1]
```diff
@@ -53,7 +53,7 @@
         interface. During an audit, a <code>Checker</code>
         informs its attached <code>AuditListeners</code> of
         six kinds of events: audit started/ended, file started/ended,
-        and logging of an error/exception.
+        and logging of an error(violation)/exception.
```
Comment (romani) [inline src/xdocs/writinglisteners.xml.vm]: it is intentional, as currently AuditListeners has only `addError` method.

File: src/xdocs/writinglisteners.xml.vm [hunk 2]
```diff
@@ -64,7 +64,7 @@
         violation logging has a message, a severity level, a message source
         such as the name of a <code>Check</code>, and file
         line and column numbers that may be relevant to the violation. The
-        notification of an exception to a <code>AuditListener</code> includes an error
+        notification of an exception to a <code>AuditListener</code> includes an violation
```
Comment (pbludov) [inline src/xdocs/writinglisteners.xml.vm]: a violation

File: src/xdocs/config_reporting.xml [hunk 3]
```diff
@@ -62,7 +62,7 @@
         The property <code>checkstyle.basedir</code>
         specifies a base directory which files are then reported
         relative to. For example, if a base directory is specified as
-        <code>c:\projects\checkstyle</code>, then an error
+        <code>c:\projects\checkstyle</code>, then an violation
```
Comment (pbludov) [inline src/xdocs/config_reporting.xml]: a violation

File: src/xdocs/writingchecks.xml [hunk 4]
```diff
@@ -685,7 +685,7 @@ public class MethodLimitCheck extends AbstractCheck
       and override the abstract
       <a href="apidocs/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.html#processFiltered-java.io.File-java.util.List-">
               <code>processFiltered(java.io.File, java.util.List)</code></a> method and you&#39;re
-      done. A very simple example could fire an error if the number of files
+      done. A very simple example could fire an violation if the number of files
```
Comment (pbludov) [inline src/xdocs/writingchecks.xml]: a violation

**Comment by romani** [comment]
fixed.

```
✔ ~/java/github/romani/checkstyle [6771-part-xdoc L|✚ 3] 
$ ag " an violation"
✘-1 
```

File: src/xdocs/writingchecks.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/writingchecks.xml]: fixed

File: src/xdocs/config_reporting.xml (hunk 3 above)
Comment (romani) [reply src/xdocs/config_reporting.xml]: fixed

File: src/xdocs/writinglisteners.xml.vm (hunk 2 above)
Comment (romani) [reply src/xdocs/writinglisteners.xml.vm]: fixed

**Pull request merged**